Add displaying disabled tests and reasons in TestKitGen

LongyuZhang opened the pull request:
- integrated disabled tests information in test result summary
- support printing disabled tests and reasons

[ci skip]

Signed-off-by: Longyu Zhang <longyu.zhang@ibm.com>

LongyuZhang commented:
reviewers: @llxia @renfeiw

llxia commented on test/README.md:
````diff
@@ -117,16 +117,25 @@ Supported groups  are `regular|native`
 
 ```    
-    make _testA
+    make _testA    // testA has <disbaled> tag in playlist.xml  
````
Please use make syntax `#` for comment

llxia commented on test/README.md:
```diff
@@ -164,17 +173,21 @@ At the end of each run, test results summary will be printed:
-    TOTAL: 84   EXECUTED: 84   PASSED: 82   FAILED: 2   SKIPPED: 0
-    ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+    TOTAL: 91   EXECUTED: 84   PASSED: 82   FAILED: 2   SKIPPED: 0   DISABLED: 7
+    +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
```
- I feel it may make more sense to have it displayed in order of DISABLED, PASSED and FAILED
- And have DISABLED before SKIPPED (SKIPPED does not count into TOTAL, but DISABLED does)

smlambert replied:
agree

llxia commented on test/docs/OpenJ9TestUserGuide.md:
````diff
@@ -262,12 +262,18 @@ target
+        make _echo.disabled.testA
+        make _echo.disabled.sanity.functional
+        make _echo.disabled.extended
+    ```
````
There are no disabled tests lists anymore. I think we can keep the user guide's description the same as the quick start.

smlambert commented on test/README.md:
```diff
@@ -117,16 +117,25 @@ Supported groups  are `regular|native`
 ## 5) How to execute disabled tests?
 
-If a test is disabled using `<disbaled>` tag in playlist.xml, it can be executed through specifying the test target or add `disabled` in front of regular target.
+If a test is disabled using `<disbaled>` tag in playlist.xml, it can be executed through specifying the test target or adding `disabled` in front of regular target.
```
spelling disbaled should be disabled

smlambert commented on test/docs/OpenJ9TestUserGuide.md:
```diff
@@ -262,12 +262,18 @@ target
     element that you want to exclude.
 
-    For tests that are disabeld with playlist.xml, `disabled.mk` is generated when running `make -f run_configure.mk`. It contains a list of disabled test targets. These targets can be executed through specifying the test target or add `disabled` in front of regular target.
+    For tests that are disabeld with playlist.xml, disabled tests lists are generated when running `make -f run_configure.mk`. These targets can be executed through specifying the test target or add `disabled` in front of regular target.
```
spelling disabeld should be disabled

renfeiw commented on test/TestConfig/scripts/testKitGen/makeGenTool/mkgen.pl:
```diff
@@ -93,41 +91,38 @@ sub runmkgen {
-				$disabledTable{$lgKey} = [];
+	my @allDisHead = ('', 'disabled.', 'echo.disabled.');
+	foreach my $eachDisHead (@allDisHead) { 
+		$targetCount{$eachDisHead . "normal.all"} = 0;
```
When we count the targets, I think we don't need `normal.all`. We just need `all` and it includes all the disabled targets as well, then the number will match the total number in the result summary. The same logic should be applied to other targets, such as sanity, sanity.functional, etc.

renfeiw commented on test/TestConfig/scripts/testKitGen/makeGenTool/mkgen.pl:
```diff
@@ -614,7 +593,17 @@ sub writeTargets {
+				my @disabledReasons = split("[\t\n]", $test->{'disabled'});
+				foreach my $dReason (@disabledReasons) {
+					if ($dReason ne "") {
+						print $fhOut "$indent\@echo \"$dReason\"\n";
```
Add `| tee -a \$(Q)\$(TESTOUTPUT)\$(D)TestTargetResult\$(Q);\n";` in the the print out.

renfeiw commented on test/TestConfig/scripts/testKitGen/makeGenTool/mkgen.pl:
```diff
@@ -673,6 +662,29 @@ sub writeTargets {
+							my @disabledReasons = split("[\t\n]", $test->{'disabled'});
+							foreach my $dReason (@disabledReasons) {
+								if ($dReason ne "") {
+									print $fhOut "$indent\@echo \"$dReason\"\n";
```
Add `| tee -a \$(Q)\$(TESTOUTPUT)\$(D)TestTargetResult\$(Q);` in the the print out.

LongyuZhang commented:
Addressed all comments;
updated targetCount method; 
tested with AdoptOpenJDK; 
tested in sub-folders; 
tested with BUILD_LIST; 
tested on VMFarm.

llxia commented:
Jenkine test sanity all jdk8

llxia commented:
Jenkins test sanity all jdk8

llxia commented:
jenkins test sanity xlinux jdk8,jdk11

llxia commented:
jenkins test sanity all jdk8

pshipton commented:
jenkins is going to be restarted shortly. The PR builds can be restarted afterwards. See the #committers-public Slack channel.

llxia commented:
jenkins test sanity all jdk8

llxia commented:
jenkins test sanity xlinux,aix jdk11
